### PR TITLE
Add FXIOS-7382 [v119] actions to the opt in card (#16363)

### PR DESCRIPTION
**Pull request by petruSt24**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7382)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/16363)

## :bulb: Description

Added the actions to the opt in card view buttons
## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [X] Wrote unit tests and/or ensured the tests suite is passing
- [X] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [X] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Frontend/Fakespot/Views/FakespotOptInCardView.swift [hunk 1]
```diff
@@ -167,26 +167,31 @@ final class FakespotOptInCardView: UIView, ThemeApplicable {
...
     private func didTapTermsOfUse() {
         viewModel?.onTapTermsOfUse()
+        viewModel?.dismissViewController?()
     }
 
     @objc
     private func didTapPrivacyPolicy() {
         viewModel?.onTapPrivacyPolicy()
+        viewModel?.dismissViewController?()
     }
```
Comment (razvanlitianu) [inline Client/Frontend/Fakespot/Views/FakespotOptInCardView.swift:183]: I think dismissal of the view controller should be in the viewModel `onTapPrivacyPolicy` call. We want to keep views pretty dumb and comunicate back only that the action happened. Dismissal is too specific and I'm thinking it should actually be part of coordinator layer.

Comment (razvanlitianu) [reply Client/Frontend/Fakespot/Views/FakespotOptInCardView.swift:183]: Same fot the other ones.

Comment (petruSt24) [reply Client/Frontend/Fakespot/Views/FakespotOptInCardView.swift]: good catch, I forgot to delete those calls from the view

**Review by razvanlitianu — APPROVED** [review]

File: Client/Frontend/Fakespot/FakespotOptInCardViewModel.swift [hunk 2]
```diff
@@ -70,6 +70,15 @@ struct FakespotOptInCardViewModel {
     let secondaryButton: String = .Shopping.OptInCardSecondaryButtonTitle
     let secondaryButtonA11yId: String = AccessibilityIdentifiers.Shopping.OptInCard.secondaryButton
 
+    // MARK: Button Actions
+    var dismissViewController: (() -> Void)?
+    var shouldOptIn: (() -> Void)?
```
Comment (thatswinnie) [inline Client/Frontend/Fakespot/FakespotOptInCardViewModel.swift:75]: From the name I'd assume this is a boolean. How about we rename it to `onOptIn`?

Comment (petruSt24) [reply Client/Frontend/Fakespot/FakespotOptInCardViewModel.swift]: sounds good, updated

**Review by thatswinnie — APPROVED** [review]

**Pull request merged**